src/System.Runtime.Extensions/tests/System/EnvironmentTests.cs

```diff
@@ -215,6 +215,25 @@ public void GetFolderPath_Unix_PersonalIsHomeAndUserProfile()
             Assert.Equal(Environment.GetEnvironmentVariable("HOME"), Environment.GetFolderPath(Environment.SpecialFolder.UserProfile));
         }
 
+        [Theory]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]  // Tests OS-specific environment
+        [InlineData(Environment.SpecialFolder.ApplicationData)]
+        [InlineData(Environment.SpecialFolder.Desktop)]
+        [InlineData(Environment.SpecialFolder.DesktopDirectory)]
+        [InlineData(Environment.SpecialFolder.Fonts)]
+        [InlineData(Environment.SpecialFolder.MyMusic)]
+        [InlineData(Environment.SpecialFolder.MyPictures)]
+        [InlineData(Environment.SpecialFolder.MyVideos)]
+        [InlineData(Environment.SpecialFolder.Templates)]
+        public void GetFolderPath_Unix_SpecialFolderNotCreated_DoesNotThrow(Environment.SpecialFolder folder)
+        {
+            var path = Environment.GetFolderPath(folder, Environment.SpecialFolderOption.DoNotVerify);
+            Assert.False(System.IO.Directory.Exists(path));
+            path = Environment.GetFolderPath(folder, Environment.SpecialFolderOption.Create);
+            Assert.True(System.IO.Directory.Exists(path));
+            Directory.Delete(path);
```

Comment:
This seems problematic.  Presumably Environment.GetFolderPath is going to be accessing some shared location.  We might delete something that's in use, even if we only delete it if we just created it, as there could be a race condition.